### PR TITLE
Sta/romain comments retail notebook

### PR DESCRIPTION
**Pull request by SamTaz**
@rcolle made some comments on the retail notebook [here](https://github.com/atoti/notebooks/issues/17), most of which apply to all our notebooks.

I made the update on the retail notebook, and jumped on the occasion to also use our notebook utils to download data and add some delta measures to directly have the margin & price index difference in the pivot tables at the end.

### CODE REVIEW

**Comment by HuifangYeo** [comment]
Hi @SamTaz,

Just wondering if you have updated the db with CE version. I encountered the below error:
image

File: retail/pricing-simulations-around-product-classes/main.ipynb [hunk 1]
```diff
@@ -545,32 +593,25 @@
...
+    "m[\"Price Index\"] = 100 * price_index_numerator / m[\"Margin.SUM\"]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The price index measure is now created accordingly to the formula.\n",
+    "\n",
+    "We can see below that our driver products are in fact a bit more expensive than the average market price while we are quite competitive on the complementary products, which is the opposite of our goal."
```
Comment (HuifangYeo) [inline retail/pricing-simulations-around-product-classes/main.ipynb:607]: How do we decide using the price index that a product is more competitive? Is there a range e.g. > 100 is too expensive, < 95 is too cheap etc?

**Comment by rcolle** [comment]
Why would you use some notebook utils when you want a drop-in notebook that can be used as an example? Moreover, why would you require wget and download the files locally when you can directly load them using a s3 addressing scheme (`s3://bucketname/path`) with either pandas or atoti?

As mentioned in the issue, you should also remove any mention to sampling if the data is small enough.

It looks like you are still using pandas to simply read CSVs (e.g. `competitor_prices`).

When loading a store, it could be interesting to display the first 10 lines so that readers can directly see what the data looks like within the notebook.

**Comment by jbe456** [comment]
>Why would you use some notebook utils when you want a drop-in notebook that can be used as an example? Moreover, why would you require wget and download the files locally when you can directly load them using a s3 addressing scheme (s3://bucketname/path) with either pandas or atoti?

The util method achieves several goals:
- make the notebook runnable offline
- limit bandwidth usage on the S3 bucket as it costs money :heavy_dollar_sign: 
- make the notebook loads data faster for subsequent executions

**Comment by rcolle** [comment]
However that's yet another non-standard layer that clutters the code. For little to no benefits in my opinion:
- what's the advantage of making the notebook run offline? You need an initial connection to download the data, you need to install `wget` so that you can run an example notebook a second time offline? I don't buy it.
- s3 is dirty cheap. Have you computed the costs? I would assume that even if 1,000 users were running the notebooks everyday, it would cost us less than 100$.
- it's an example notebook. People run it once at most. More often than not they'll just look it up on github and start by seeing some weird notebook utils code.

File: retail/pricing-simulations-around-product-classes/main.ipynb [hunk 2]
```diff
@@ -847,17 +917,36 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 41,
+   "execution_count": 40,
    "metadata": {
     "atoti": {
      "state": {
-      "name": "",
+      "name": "Results summary",
```
[the remark this answers is not in the document]
Comment (SamTaz) [reply retail/pricing-simulations-around-product-classes/main.ipynb:924]: Green is good, red is bad.
I'll add that in the notebook comments

**Comment by jbe456** [comment]
I agree. These are relics from when we originally considered adding this notebook and data was 3.5 GB.

Simplicity should be our goal. We'll therefore:
- remove the utils
- clean up the file in advance to avoid doing pandas clean up
- use the s3 protocol to directly get the file with Atoti methods

**Comment by rcolle** [comment]
Sounds good to me.
For clarity sake, I don't mind using pandas if some data cleanup is involved (it could even be argued that it's a showcase of our integration, although the simplicity argument is also valid). I just want to avoid it when we're directly reading a csv file "as-is".

**Comment by SamTaz** [comment]
Hello Huifang,

> We can see below that this represents a gain of 10 million per year, while increasing the index dramatically from **98.52** to 108.37 <-- Not sure if the table value was correct in the original version and changed in this version. Now it's showing 9.85 istead of 98.52.

In the base scenario the price index is 98.52, 108.37 - 98.52 = 9.85, the delta measure is just the difference.

> Also, in the "Result Summary" table, is the Margin Delta at Optimized price index a summation of all the 3 product category's Margin Delta? It doesn't seem to add up.

Looks like you selected only 2 cells in the picture. If you select all 3 cells it does add up.

**Pull request merged**